enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/APIKeyAuthenticator.java

```diff
@@ -74,26 +86,50 @@ private String getAPIKeyFromRequest(RequestContext requestContext) {
                 .getApiKeyInternalHeader().toLowerCase());
     }
 
-    @Override
-    protected String retrieveTokenFromRequestCtx(RequestContext requestContext) throws APISecurityException {
-
+    private JSONObject getDecodedAPIKeyData(String apiKeyHeaderValue) throws APISecurityException {
         try {
-            String apiKeyHeaderValue = getAPIKeyFromRequest(requestContext).trim();
             // Skipping the prefix(`chk_`) and checksum.
             String apiKeyData = apiKeyHeaderValue.substring(4, apiKeyHeaderValue.length() - 6);
             // Base 64 decode key data.
             String decodedKeyData = new String(Base64.getDecoder().decode(apiKeyData));
             // Convert data into JSON.
-            JSONObject jsonObject = (JSONObject) JSONValue.parse(decodedKeyData);
-            // Extracting the jwt token.
-            return jsonObject.getAsString(APIKeyConstants.API_KEY_JSON_KEY);
+            return (JSONObject) JSONValue.parse(decodedKeyData);
         } catch (Exception e) {
             throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
                     APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
                     APISecurityConstants.API_AUTH_INVALID_CREDENTIALS_MESSAGE);
         }
     }
 
+    @Override
+    protected String retrieveTokenFromRequestCtx(RequestContext requestContext) throws APISecurityException {
+
+        String apiKey = getAPIKeyFromRequest(requestContext);
+        if (!APIKeyUtils.isValidAPIKey(apiKey)) {
+            throw new APISecurityException(APIConstants.StatusCodes.UNAUTHENTICATED.getCode(),
+                    APISecurityConstants.API_AUTH_INVALID_CREDENTIALS,
+                    APISecurityConstants.API_AUTH_INVALID_CREDENTIALS_MESSAGE);
+        }
+        String keyHash = APIKeyUtils.generateAPIKeyHash(apiKey);
```

Comment:
This method also could generate IllegalArgumentException if not a valid base64 payload.